install_sct

```diff
@@ -13,14 +13,13 @@
 # installation is successful.
 #
 # USAGE
-#   ./install_sct [-h] [-y] [-d] [-b] [-v]
+#   ./install_sct [-h] [-y] [-c] [-v]
 #   ./install_sct
 #
 # OPTIONS
 #  -h   Show this help
 #  -y   Install without interruption with 'yes' as default answer
-#  -d   Prevent the (re)-installation of the data/ directory
-#  -b   Prevent the (re)-installation of the SCT binaries files
+#  -c   Clean previous installation (by default, data/ et al. are not redownloaded)
```

Comment:
correct me if i'm wrong but what if someone installs SCT, then reinstall it 3 months later: with this new default, the potentially-updated data et al. won't be reinstalled, right?  i anticipate this could be a problem (e.g. we fix something with the data, and ppl will miss the fix).

Reply:
Yes, that's exactly my question to you pretty much.

The root there is that `sct_download_data` doesn't support versioning. We are versioning our downloads thanks to osf forcing that on us and now distriploy, but we're not having users record that version anywhere. Until we sort that out (#1526 !), I reasoned that cheating this a little bit could be okay if:

- users tend to install from releases, which creates more isolated environments under `~/sct_$VER`, hence every install downloads `bin/` and `data/` fresh
- we tend not to update the data very often

Reply:
What if I added to this PR basic version tracking to `sct_download_data`? I could have it write a `.source` file in each folder with the URL of the .zip it came from, and then we can compare that to see if it's changed; a bit hacky but it would work for now until we do something better.

Or I could try to compare `pkg_resources.get_distribution('spinalcordtoolbox').version` against the version of what we're trying to install, and set `CLEAN` automatically if it's changed; except that for the `.git` (aka `SCT_INSTALL_TYPE="in-place"`) version, this always returns "dev". If we put, like, `5.0.1-dev` in there instead.

Reply:
> - users tend to install from releases, which creates more isolated environments under ~/sct_$VER, hence every install downloads bin/ and data/ fresh

this is a valid arguments, but in the past i remember spending days trying to figure out why external users and students in the lab had unexpected behaviour, after realizing that what was different was the data (ie it did not get reinstalled, used an old or corrupted version, etc.). So we made the drastic move: with `install_sct` we make 100% sure we know what data are installed.

my approach would be: for ppl who don't know what they are doing, default behaviour should be "brute force clean install". For ppl who know what they are doing (e.g. you), you can use the flag when you deem relevant to not re-download the data systematically (e.g. for CI).

Out of curiosity: what were the scenarii where you felt clean reinstall was superfluous? 

> - we tend not to update the data very often

This is also a valid argument, but: maybe in 6 months from now, we will update deep learning models every week. Also: the fact that we don't update data very often implies that we _do_ sometimes update the data. And when that happens, everyone (internal and external users) will need to know about the existence of the "-c" flag, and they need to know that they need to use it at that precise moment. I anticipate not 100% of users will do it, it will cause unexpected behaviours, and more white hair for me.

Reply:
I thought maybe you'd say something like that. I hear you.

Okay, so what if I changed this PR to flip it: add a `-C` flag that means DON'T clean instead? It would still combine the `-d` and `-b` flags into one and make sure to cover the same for the deepseg models, but also apply to conda/pip, and devs can use `./install_sct -C` for day to day testing, and `./install_sct` (or just wait for CI) to catch anything that slips through that.

There should never be any reason to redownload conda/pip. conda doesn't change that fast -- it's point is to give a stable platform and plus we explicitly use conda to install a specific version of python, and pip understands versioning so if something does change it knows to reinstall a package. Redownloading sklearn+tensorflow+torch (~300MB) just because, say, we're not sure if xlwt or matplotlib is the right version seems superfluous; and no, not a big deal our users, but a nuisance for us devs.

Now, we might have some trouble with this at first because we don't pin our versions [except for releases](https://github.com/neuropoly/spinalcordtoolbox/blob/6172cd22feeb0f64ac1bd68bce3f89f7109e68b8/install_sct#L558-L560) or [specific packages we've had trouble with](https://github.com/neuropoly/spinalcordtoolbox/blob/6172cd22feeb0f64ac1bd68bce3f89f7109e68b8/requirements.txt#L7-L8), but if we used [`the `~=` syntax](https://www.python.org/dev/peps/pep-0440/#compatible-release) for everything like we do here

https://github.com/neuropoly/spinalcordtoolbox/blob/6172cd22feeb0f64ac1bd68bce3f89f7109e68b8/requirements.txt#L6

then reusing the venv should be safe.

I could go through an add `~=` to the current requirements.txt based on what versions we currently use as part of this PR. That would be a step towards making #1526 work.

Reply:
> I could go through an add ~= to the current requirements.txt based on what versions we currently use as part of this PR. That would be a step towards making #1526 work.

I shared some related thoughts about using `~=` here: https://github.com/neuropoly/axondeepseg/issues/411#issuecomment-740145463. At the time, I was against using `~=` for everything, in favor of keeping our development requirements more open. 

> Now, we might have some trouble with this at first because we don't pin our versions

Could you clarify what trouble you think we might run into? I might be off-base with being against version-pinning.

Reply:
I've inverted the `-c` -> `-C` in  80657657afb2127ecaf7d2ab29ddd1f0549931e3

<details><summary>With cleaning: ~4 minutes</summary>

```
$ time ./install_sct -y

 non-interactive mode



*******************************
* Welcome to SCT installation *
*******************************



Checking OS type and version...

Linux requiem 5.9.14-arch1-1 #1 SMP PREEMPT Sat, 12 Dec 2020 14:37:12 +0000 x86_64 GNU/Linux

Checking requirements...


OK!


SCT version ......... dev
Installation type ... in-place
Operating system .... linux (unknown)
Shell config ........ /home/kousu/.bashrc

--> Crash reports will not be sent.


SCT will be installed here: [/home/kousu/src/neuropoly/spinalcordtoolbox]


Skipping copy of source files (source and destination folders are the same)


Removing sct and isct softlink from /home/kousu/src/neuropoly/spinalcordtoolbox/bin


rm -rf /home/kousu/src/neuropoly/spinalcordtoolbox/python


Installing conda...


mkdir -p /home/kousu/src/neuropoly/spinalcordtoolbox/python


wget -O /tmp/tmp.14bcijo9EP/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh

--2020-12-30 19:37:18--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8203, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 94235922 (90M) [application/x-sh]
Saving to: ‘/tmp/tmp.14bcijo9EP/miniconda.sh’

/tmp/tmp.14bcijo9EP/miniconda.sh                 100%[========================================================================================================>]  89.87M  1.41MB/s    in 42s     

2020-12-30 19:38:00 (2.13 MB/s) - ‘/tmp/tmp.14bcijo9EP/miniconda.sh’ saved [94235922/94235922]


bash /tmp/tmp.14bcijo9EP/miniconda.sh -p /home/kousu/src/neuropoly/spinalcordtoolbox/python -b -f

PREFIX=/home/kousu/src/neuropoly/spinalcordtoolbox/python
Unpacking payload ...
Collecting package metadata (current_repodata.json): done                                                                                                                                         
Solving environment: done

## Package Plan ##

  environment location: /home/kousu/src/neuropoly/spinalcordtoolbox/python

  added / updated specs:
    - _libgcc_mutex==0.1=main
    - brotlipy==0.7.0=py38h27cfd23_1003
    - ca-certificates==2020.10.14=0
    - certifi==2020.6.20=pyhd3eb1b0_3
    - cffi==1.14.3=py38h261ae71_2
    - chardet==3.0.4=py38h06a4308_1003
    - conda-package-handling==1.7.2=py38h03888b9_0
    - conda==4.9.2=py38h06a4308_0
    - cryptography==3.2.1=py38h3c74f83_1
    - idna==2.10=py_0
    - ld_impl_linux-64==2.33.1=h53a641e_7
    - libedit==3.1.20191231=h14c3975_1
    - libffi==3.3=he6710b0_2
    - libgcc-ng==9.1.0=hdf63c60_0
    - libstdcxx-ng==9.1.0=hdf63c60_0
    - ncurses==6.2=he6710b0_1
    - openssl==1.1.1h=h7b6447c_0
    - pip==20.2.4=py38h06a4308_0
    - pycosat==0.6.3=py38h7b6447c_1
    - pycparser==2.20=py_2
    - pyopenssl==19.1.0=pyhd3eb1b0_1
    - pysocks==1.7.1=py38h06a4308_0
    - python==3.8.5=h7579374_1
    - readline==8.0=h7b6447c_0
    - requests==2.24.0=py_0
    - ruamel_yaml==0.15.87=py38h7b6447c_1
    - setuptools==50.3.1=py38h06a4308_1
    - six==1.15.0=py38h06a4308_0
    - sqlite==3.33.0=h62c20be_0
    - tk==8.6.10=hbc83047_0
    - tqdm==4.51.0=pyhd3eb1b0_0
    - urllib3==1.25.11=py_0
    - wheel==0.35.1=pyhd3eb1b0_0
    - xz==5.2.5=h7b6447c_0
    - yaml==0.2.5=h7b6447c_0
    - zlib==1.2.11=h7b6447c_3


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py38h27cfd23_1003
  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.10.14-0
  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3
  cffi               pkgs/main/linux-64::cffi-1.14.3-py38h261ae71_2
  chardet            pkgs/main/linux-64::chardet-3.0.4-py38h06a4308_1003
  conda              pkgs/main/linux-64::conda-4.9.2-py38h06a4308_0
  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.7.2-py38h03888b9_0
  cryptography       pkgs/main/linux-64::cryptography-3.2.1-py38h3c74f83_1
  idna               pkgs/main/noarch::idna-2.10-py_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1h-h7b6447c_0
  pip                pkgs/main/linux-64::pip-20.2.4-py38h06a4308_0
  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py38h7b6447c_1
  pycparser          pkgs/main/noarch::pycparser-2.20-py_2
  pyopenssl          pkgs/main/noarch::pyopenssl-19.1.0-pyhd3eb1b0_1
  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py38h06a4308_0
  python             pkgs/main/linux-64::python-3.8.5-h7579374_1
  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0
  requests           pkgs/main/noarch::requests-2.24.0-py_0
  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py38h7b6447c_1
  setuptools         pkgs/main/linux-64::setuptools-50.3.1-py38h06a4308_1
  six                pkgs/main/linux-64::six-1.15.0-py38h06a4308_0
  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  tqdm               pkgs/main/noarch::tqdm-4.51.0-pyhd3eb1b0_0
  urllib3            pkgs/main/noarch::urllib3-1.25.11-py_0
  wheel              pkgs/main/noarch::wheel-0.35.1-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3


Preparing transaction: done
Executing transaction: done
installation finished.
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/kousu/src/neuropoly/spinalcordtoolbox/python/envs/venv_sct

  added / updated specs:
    - python=3.6


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.12.8  |       h06a4308_0         121 KB
    certifi-2020.12.5          |   py36h06a4308_0         140 KB
    openssl-1.1.1i             |       h27cfd23_0         2.5 MB
    pip-20.3.3                 |   py36h06a4308_0         1.8 MB
    python-3.6.12              |       hcff3b4d_2        29.7 MB
    setuptools-51.0.0          |   py36h06a4308_2         712 KB
    wheel-0.36.2               |     pyhd3eb1b0_0          33 KB
    ------------------------------------------------------------
                                           Total:        35.0 MB

The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main
  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.12.8-h06a4308_0
  certifi            pkgs/main/linux-64::certifi-2020.12.5-py36h06a4308_0
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7
  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1
  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0
  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1
  openssl            pkgs/main/linux-64::openssl-1.1.1i-h27cfd23_0
  pip                pkgs/main/linux-64::pip-20.3.3-py36h06a4308_0
  python             pkgs/main/linux-64::python-3.6.12-hcff3b4d_2
  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0
  setuptools         pkgs/main/linux-64::setuptools-51.0.0-py36h06a4308_2
  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0
  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0
  wheel              pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0
  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0
  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3



Downloading and Extracting Packages
certifi-2020.12.5    | 140 KB    | ####################################################################################################################################################### | 100% 
wheel-0.36.2         | 33 KB     | ####################################################################################################################################################### | 100% 
python-3.6.12        | 29.7 MB   | ####################################################################################################################################################### | 100% 
setuptools-51.0.0    | 712 KB    | ####################################################################################################################################################### | 100% 
pip-20.3.3           | 1.8 MB    | ####################################################################################################################################################### | 100% 
openssl-1.1.1i       | 2.5 MB    | ####################################################################################################################################################### | 100% 
ca-certificates-2020 | 121 KB    | ####################################################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate venv_sct
#
# To deactivate an active environment, use
#
#     $ conda deactivate


Installing Python dependencies...


Using requirements.txt (git installation)

Looking in links: https://download.pytorch.org/whl/cpu/torch_stable.html
Ignoring torch: markers 'sys_platform == "darwin"' don't match your environment
Ignoring torchvision: markers 'sys_platform == "darwin"' don't match your environment
Collecting ivadomed==2.6.1
  Using cached ivadomed-2.6.1-py3-none-any.whl (145 kB)
Collecting h5py~=2.10.0
  Using cached h5py-2.10.0-cp36-cp36m-manylinux1_x86_64.whl (2.9 MB)
Collecting Keras==2.1.5
  Using cached Keras-2.1.5-py2.py3-none-any.whl (334 kB)
Collecting onnxruntime==1.4.0
  Using cached onnxruntime-1.4.0-cp36-cp36m-manylinux2010_x86_64.whl (4.4 MB)
Collecting pyqt5==5.11.3
  Using cached PyQt5-5.11.3-5.11.2-cp35.cp36.cp37.cp38-abi3-manylinux1_x86_64.whl (117.8 MB)
Collecting tensorboard==1.14.0
  Using cached tensorboard-1.14.0-py3-none-any.whl (3.1 MB)
Requirement already satisfied: setuptools>=41.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (51.0.0.post20201207)
Requirement already satisfied: wheel>=0.26 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (0.36.2)
Collecting tensorflow==1.5.0
  Using cached tensorflow-1.5.0-cp36-cp36m-manylinux1_x86_64.whl (44.4 MB)
Collecting torch==1.5.0+cpu
  Using cached https://download.pytorch.org/whl/cpu/torch-1.5.0%2Bcpu-cp36-cp36m-linux_x86_64.whl (127.3 MB)
Collecting torchvision==0.6.0+cpu
  Using cached https://download.pytorch.org/whl/cpu/torchvision-0.6.0%2Bcpu-cp36-cp36m-linux_x86_64.whl (5.7 MB)
Collecting matplotlib
  Using cached matplotlib-3.3.3-cp36-cp36m-manylinux1_x86_64.whl (11.6 MB)
Collecting numpy
  Using cached numpy-1.19.4-cp36-cp36m-manylinux2010_x86_64.whl (14.5 MB)
Collecting scikit-learn
  Using cached scikit_learn-0.24.0-cp36-cp36m-manylinux2010_x86_64.whl (22.2 MB)
Collecting scipy
  Using cached scipy-1.5.4-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)
Collecting tqdm
  Using cached tqdm-4.55.0-py2.py3-none-any.whl (68 kB)
Collecting absl-py>=0.4
  Using cached absl_py-0.11.0-py3-none-any.whl (127 kB)
Collecting bids-neuropoly>=0.2
  Using cached bids_neuropoly-0.2-py3-none-any.whl (3.1 kB)
Collecting nibabel
  Using cached nibabel-3.2.1-py3-none-any.whl (3.3 MB)
Collecting pandas
  Using cached pandas-1.1.5-cp36-cp36m-manylinux1_x86_64.whl (9.5 MB)
Collecting cycler>=0.10
  Using cached cycler-0.10.0-py2.py3-none-any.whl (6.5 kB)
Collecting grpcio>=1.6.3
  Using cached grpcio-1.34.0-cp36-cp36m-manylinux2014_x86_64.whl (4.0 MB)
Collecting joblib
  Using cached joblib-1.0.0-py3-none-any.whl (302 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.3.1-cp36-cp36m-manylinux1_x86_64.whl (1.1 MB)
Collecting markdown>=2.6.8
  Using cached Markdown-3.3.3-py3-none-any.whl (96 kB)
Collecting packaging>=14.3
  Using cached packaging-20.8-py2.py3-none-any.whl (39 kB)
Collecting pillow>=7.0.0
  Using cached Pillow-8.0.1-cp36-cp36m-manylinux1_x86_64.whl (2.2 MB)
Collecting protobuf
  Using cached protobuf-3.14.0-cp36-cp36m-manylinux1_x86_64.whl (1.0 MB)
Collecting pybids>=0.12.4
  Using cached pybids-0.12.4-py2.py3-none-any.whl (3.2 MB)
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting PyQt5_sip<4.20,>=4.19.11
  Using cached PyQt5_sip-4.19.19-cp36-cp36m-manylinux1_x86_64.whl (67 kB)
Collecting python-dateutil>=2.1
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting pytz>=2017.2
  Using cached pytz-2020.5-py2.py3-none-any.whl (510 kB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting sqlalchemy<1.4.0.dev0
  Using cached SQLAlchemy-1.3.22-cp36-cp36m-manylinux2010_x86_64.whl (1.3 MB)
Collecting tensorflow-tensorboard<1.6.0,>=1.5.0
  Using cached tensorflow_tensorboard-1.5.1-py3-none-any.whl (3.0 MB)
Collecting bleach==1.5.0
  Using cached bleach-1.5.0-py2.py3-none-any.whl (17 kB)
Collecting html5lib==0.9999999
  Using cached html5lib-0.9999999-py3-none-any.whl
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-2.1.0-py3-none-any.whl (12 kB)
Collecting werkzeug>=0.11.15
  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting colored
  Using cached colored-1.4.2-py3-none-any.whl
Collecting dipy
  Using cached dipy-1.3.0-cp36-cp36m-manylinux2010_x86_64.whl (7.7 MB)
Collecting futures
  Using cached futures-3.1.1-py3-none-any.whl (2.8 kB)
Collecting psutil
  Using cached psutil-5.8.0-cp36-cp36m-manylinux2010_x86_64.whl (291 kB)
Collecting pytest
  Using cached pytest-6.2.1-py3-none-any.whl (279 kB)
Collecting attrs>=19.2.0
  Using cached attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting importlib-metadata
  Using cached importlib_metadata-3.3.0-py3-none-any.whl (10 kB)
Collecting pluggy<1.0.0a1,>=0.12
  Using cached pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting py>=1.8.2
  Using cached py-1.10.0-py2.py3-none-any.whl (97 kB)
Collecting typing-extensions>=3.6.4
  Using cached typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
Collecting zipp>=0.5
  Using cached zipp-3.4.0-py3-none-any.whl (5.2 kB)
Collecting pytest_console_scripts
  Using cached pytest_console_scripts-1.1.0-py3-none-any.whl
Collecting mock>=2.0.0
  Using cached mock-4.0.3-py3-none-any.whl (28 kB)
Collecting pytest-cov
  Using cached pytest_cov-2.10.1-py2.py3-none-any.whl (19 kB)
Collecting coverage>=4.4
  Using cached coverage-5.3.1-cp36-cp36m-manylinux2010_x86_64.whl (241 kB)
Collecting pytest-xdist
  Using cached pytest_xdist-2.2.0-py3-none-any.whl (36 kB)
Collecting execnet>=1.1
  Using cached execnet-1.7.1-py2.py3-none-any.whl (39 kB)
Collecting apipkg>=1.4
  Using cached apipkg-1.5-py2.py3-none-any.whl (4.9 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)
Requirement already satisfied: certifi>=2017.4.17 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests->-r requirements.txt (line 22)) (2020.12.5)
Collecting chardet<5,>=3.0.2
  Using cached chardet-4.0.0-py2.py3-none-any.whl (178 kB)
Collecting idna<3,>=2.5
  Using cached idna-2.10-py2.py3-none-any.whl (58 kB)
Collecting urllib3[secure]
  Using cached urllib3-1.26.2-py2.py3-none-any.whl (136 kB)
Collecting requirements-parser
  Using cached requirements_parser-0.2.0-py3-none-any.whl
Collecting scikit-image
  Using cached scikit_image-0.17.2-cp36-cp36m-manylinux1_x86_64.whl (12.4 MB)
Collecting imageio>=2.3.0
  Using cached imageio-2.9.0-py3-none-any.whl (3.3 MB)
Collecting networkx>=2.0
  Using cached networkx-2.5-py3-none-any.whl (1.6 MB)
Collecting decorator>=4.3.0
  Using cached decorator-4.4.2-py2.py3-none-any.whl (9.2 kB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.1.1-cp36-cp36m-manylinux1_x86_64.whl (4.4 MB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2020.9.3-py3-none-any.whl (148 kB)
Collecting transforms3d
  Using cached transforms3d-0.3.1-py3-none-any.whl
Collecting cryptography>=1.3.4
  Using cached cryptography-3.3.1-cp36-abi3-manylinux2010_x86_64.whl (2.6 MB)
Collecting cffi>=1.12
  Using cached cffi-1.14.4-cp36-cp36m-manylinux1_x86_64.whl (401 kB)
Collecting pyOpenSSL>=0.14
  Using cached pyOpenSSL-20.0.1-py2.py3-none-any.whl (54 kB)
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting bids-validator
  Using cached bids_validator-1.5.8-py2.py3-none-any.whl (15 kB)
Collecting click
  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting num2words
  Using cached num2words-0.5.10-py3-none-any.whl (101 kB)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2-py2.py3-none-any.whl
Collecting patsy
  Using cached patsy-0.5.1-py2.py3-none-any.whl (231 kB)
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Collecting pytest-forked
  Using cached pytest_forked-1.3.0-py2.py3-none-any.whl (4.7 kB)
Collecting pyyaml
  Using cached PyYAML-5.3.1-cp36-cp36m-linux_x86_64.whl
Collecting seaborn
  Using cached seaborn-0.11.1-py3-none-any.whl (285 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Installing collected packages: zipp, typing-extensions, six, pyparsing, pytz, python-dateutil, pycparser, packaging, numpy, importlib-metadata, docopt, toml, sqlalchemy, scipy, py, pluggy, pillow, patsy, pandas, num2words, nibabel, kiwisolver, iniconfig, html5lib, future, decorator, cycler, click, cffi, bids-validator, attrs, werkzeug, torch, tifffile, threadpoolctl, PyWavelets, pytest, pybids, protobuf, networkx, matplotlib, markdown, joblib, imageio, grpcio, cryptography, bleach, apipkg, absl-py, urllib3, tqdm, torchvision, tensorflow-tensorboard, tensorboard, seaborn, scikit-learn, scikit-image, pyyaml, pytest-forked, PyQt5-sip, pyOpenSSL, onnxruntime, mock, idna, h5py, execnet, coverage, chardet, bids-neuropoly, xlwt, transforms3d, tensorflow, requirements-parser, requests, raven, pytest-xdist, pytest-cov, pytest-console-scripts, pyqt5, psutil, Keras, ivadomed, futures, dipy, colored
Successfully installed Keras-2.1.5 PyQt5-sip-4.19.19 PyWavelets-1.1.1 absl-py-0.11.0 apipkg-1.5 attrs-20.3.0 bids-neuropoly-0.2 bids-validator-1.5.8 bleach-1.5.0 cffi-1.14.4 chardet-4.0.0 click-7.1.2 colored-1.4.2 coverage-5.3.1 cryptography-3.3.1 cycler-0.10.0 decorator-4.4.2 dipy-1.3.0 docopt-0.6.2 execnet-1.7.1 future-0.18.2 futures-3.1.1 grpcio-1.34.0 h5py-2.10.0 html5lib-0.9999999 idna-2.10 imageio-2.9.0 importlib-metadata-3.3.0 iniconfig-1.1.1 ivadomed-2.6.1 joblib-1.0.0 kiwisolver-1.3.1 markdown-3.3.3 matplotlib-3.3.3 mock-4.0.3 networkx-2.5 nibabel-3.2.1 num2words-0.5.10 numpy-1.19.4 onnxruntime-1.4.0 packaging-20.8 pandas-1.1.5 patsy-0.5.1 pillow-8.0.1 pluggy-0.13.1 protobuf-3.14.0 psutil-5.8.0 py-1.10.0 pyOpenSSL-20.0.1 pybids-0.12.4 pycparser-2.20 pyparsing-2.4.7 pyqt5-5.11.3 pytest-6.2.1 pytest-console-scripts-1.1.0 pytest-cov-2.10.1 pytest-forked-1.3.0 pytest-xdist-2.2.0 python-dateutil-2.8.1 pytz-2020.5 pyyaml-5.3.1 raven-6.10.0 requests-2.25.1 requirements-parser-0.2.0 scikit-image-0.17.2 scikit-learn-0.24.0 scipy-1.5.4 seaborn-0.11.1 six-1.15.0 sqlalchemy-1.3.22 tensorboard-1.14.0 tensorflow-1.5.0 tensorflow-tensorboard-1.5.1 threadpoolctl-2.1.0 tifffile-2020.9.3 toml-0.10.2 torch-1.5.0+cpu torchvision-0.6.0+cpu tqdm-4.55.0 transforms3d-0.3.1 typing-extensions-3.7.4.3 urllib3-1.26.2 werkzeug-1.0.1 xlwt-1.3.0 zipp-3.4.0
Found existing installation: tensorflow-tensorboard 1.5.1
Uninstalling tensorflow-tensorboard-1.5.1:
  Successfully uninstalled tensorflow-tensorboard-1.5.1
Found existing installation: tensorboard 1.14.0
Uninstalling tensorboard-1.14.0:
  Successfully uninstalled tensorboard-1.14.0
Collecting tensorboard
  Using cached tensorboard-2.4.0-py3-none-any.whl (10.6 MB)
Requirement already satisfied: setuptools>=41.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (51.0.0.post20201207)
Requirement already satisfied: markdown>=2.6.8 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (3.3.3)
Requirement already satisfied: absl-py>=0.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (0.11.0)
Requirement already satisfied: protobuf>=3.6.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (3.14.0)
Requirement already satisfied: wheel>=0.26 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (0.36.2)
Requirement already satisfied: requests<3,>=2.21.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (2.25.1)
Requirement already satisfied: werkzeug>=0.11.15 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.0.1)
Requirement already satisfied: numpy>=1.12.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.19.4)
Requirement already satisfied: grpcio>=1.24.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.34.0)
Requirement already satisfied: six>=1.10.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.15.0)
Collecting google-auth<2,>=1.6.3
  Using cached google_auth-1.24.0-py2.py3-none-any.whl (114 kB)
Collecting cachetools<5.0,>=2.0.0
  Using cached cachetools-4.2.0-py3-none-any.whl (12 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.2-py2.py3-none-any.whl (18 kB)
Requirement already satisfied: importlib-metadata in ./python/envs/venv_sct/lib/python3.6/site-packages (from markdown>=2.6.8->tensorboard) (3.3.0)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Requirement already satisfied: idna<3,>=2.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (2020.12.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (1.26.2)
Requirement already satisfied: chardet<5,>=3.0.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (4.0.0)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.1.0-py2.py3-none-any.whl (147 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.6-py3-none-any.whl (47 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.7.0-py3-none-any.whl (779 kB)
Requirement already satisfied: typing-extensions>=3.6.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard) (3.7.4.3)
Requirement already satisfied: zipp>=0.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard) (3.4.0)
Installing collected packages: pyasn1, rsa, pyasn1-modules, oauthlib, cachetools, requests-oauthlib, google-auth, tensorboard-plugin-wit, google-auth-oauthlib, tensorboard
Successfully installed cachetools-4.2.0 google-auth-1.24.0 google-auth-oauthlib-0.4.2 oauthlib-3.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 requests-oauthlib-1.3.0 rsa-4.6 tensorboard-2.4.0 tensorboard-plugin-wit-1.7.0

Installing spinalcordtoolbox...

Obtaining file:///home/kousu/src/neuropoly/spinalcordtoolbox
Installing collected packages: spinalcordtoolbox
  Running setup.py develop for spinalcordtoolbox
Successfully installed spinalcordtoolbox

Creating launchers for Python scripts...


 SCT binaries will not be reinstalled


All requirements installed!


rm -rf /home/kousu/src/neuropoly/spinalcordtoolbox/data


Installing data...


mkdir -p /home/kousu/src/neuropoly/spinalcordtoolbox/data


sct_download_data -d PAM50 -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/PAM50


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d PAM50 -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/PAM50
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/PAM50'
Trying URL: https://github.com/sct-data/PAM50/releases/download/r20201104/PAM50-r20201104.zip
Downloading: PAM50-r20201104.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70.9M/70.9M [00:26<00:00, 2.65MB/s]
Creating temporary folder (/tmp/sct-20201230194101.441025-48fajuw8)
Unzip data to: /tmp/sct-20201230194101.441025-48fajuw8
Removing temporary folders...
Done!


sct_download_data -d gm_model -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/gm_model


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d gm_model -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/gm_model
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/gm_model'
Trying URL: https://osf.io/ugscu/?action=download
Downloading: 20160922_gm_model.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3M/10.3M [00:03<00:00, 2.77MB/s]
Creating temporary folder (/tmp/sct-20201230194107.473632-51znso0u)
Unzip data to: /tmp/sct-20201230194107.473632-51znso0u
Removing temporary folders...
Done!


sct_download_data -d optic_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/optic_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d optic_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/optic_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/optic_models'
Trying URL: https://github.com/sct-data/optic_models/releases/download/r20170413/20170413_optic_models.zip
Downloading: 20170413_optic_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 17.7k/17.7k [00:00<00:00, 52.1MB/s]
Creating temporary folder (/tmp/sct-20201230194108.425534-re2w4q_2)
Unzip data to: /tmp/sct-20201230194108.425534-re2w4q_2
Removing temporary folders...
Done!


sct_download_data -d pmj_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/pmj_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d pmj_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/pmj_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/pmj_models'
Trying URL: https://github.com/sct-data/pmj_models/releases/download/r20170922/20170922_pmj_models.zip
Downloading: 20170922_pmj_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8.87k/8.87k [00:00<00:00, 15.9MB/s]
Creating temporary folder (/tmp/sct-20201230194109.270246-k4gjbuac)
Unzip data to: /tmp/sct-20201230194109.270246-k4gjbuac
Removing temporary folders...
Done!


sct_download_data -d deepseg_sc_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_sc_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d deepseg_sc_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_sc_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_sc_models'
Trying URL: https://github.com/sct-data/deepseg_sc_models/releases/download/r20180610/20180610_deepseg_sc_models.zip
Downloading: 20180610_deepseg_sc_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 58.7M/58.7M [00:22<00:00, 2.58MB/s]
Creating temporary folder (/tmp/sct-20201230194132.922769-k__edyd3)
Unzip data to: /tmp/sct-20201230194132.922769-k__edyd3
Removing temporary folders...
Done!


sct_download_data -d deepseg_gm_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_gm_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d deepseg_gm_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_gm_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_gm_models'
Trying URL: https://github.com/sct-data/deepseg_gm_models/releases/download/r20180205/20180205_deepseg_gm_models.zip
Downloading: 20180205_deepseg_gm_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2.28M/2.28M [00:00<00:00, 2.62MB/s]
Creating temporary folder (/tmp/sct-20201230194135.347059-ey_a1dci)
Unzip data to: /tmp/sct-20201230194135.347059-ey_a1dci
Removing temporary folders...
Done!


sct_download_data -d deepseg_lesion_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_lesion_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d deepseg_lesion_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_lesion_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_lesion_models'
Trying URL: https://github.com/sct-data/deepseg_lesion_models/releases/download/r20180613/20180613_deepseg_lesion_models.zip
Downloading: 20180613_deepseg_lesion_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 16.0M/16.0M [00:07<00:00, 2.16MB/s]
Creating temporary folder (/tmp/sct-20201230194143.957448-_ay_8rvm)
Unzip data to: /tmp/sct-20201230194143.957448-_ay_8rvm
Removing temporary folders...
Done!


sct_download_data -d c2c3_disc_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/c2c3_disc_models


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_download_data -d c2c3_disc_models -o /home/kousu/src/neuropoly/spinalcordtoolbox/data/c2c3_disc_models
--

Removing existing destination folder '/home/kousu/src/neuropoly/spinalcordtoolbox/data/c2c3_disc_models'
Trying URL: https://github.com/sct-data/c2c3_disc_models/releases/download/r20190117/20190117_c2c3_disc_models.zip
Downloading: 20190117_c2c3_disc_models.zip
Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 8.94k/8.94k [00:00<00:00, 32.4MB/s]
Creating temporary folder (/tmp/sct-20201230194144.995975-w_3o33gj)
Unzip data to: /tmp/sct-20201230194144.995975-w_3o33gj
Removing temporary folders...
Done!


rm -rf /home/kousu/src/neuropoly/spinalcordtoolbox/data/deepseg_models


Install deep learning models

Status: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18.0M/18.0M [00:06<00:00, 2.61MB/s]

Validate installation...


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_check_dependencies 
--

SCT info:
- version: git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*
- path: /home/kousu/src/neuropoly/spinalcordtoolbox
OS: linux (Linux-5.9.14-arch1-1-x86_64-with-arch)
CPU cores: Available: 4, Used by ITK functions: 4
RAM: Total: 15710MB, Used: 1884MB, Available: 11925MB
Check Python executable.............................[OK]
  Using bundled python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56) 
[GCC 7.3.0] at /home/kousu/src/neuropoly/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.2)
Check if dipy is installed..........................[OK] (1.3.0)
Check if futures is installed.......................[OK]
Check if h5py is installed..........................[OK] (2.10.0)
Check if Keras (2.1.5) is installed.................[OK] (2.1.5)
Check if ivadomed (2.6.1) is installed..............[OK] (2.6.1)
Check if matplotlib is installed....................[OK] (3.3.3)
Check if nibabel is installed.......................[OK] (3.2.1)
Check if numpy is installed.........................[OK] (1.19.4)
Check if onnxruntime (1.4.0) is installed...........[OK] (1.4.0)
Check if pandas is installed........................[OK] (1.1.5)
Check if psutil is installed........................[OK] (5.8.0)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.2.1)
Check if pytest-cov is installed....................[OK] (__version__ = '2.10.1')
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.25.1)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.5.4)
Check if scikit-image is installed..................[OK] (0.17.2)
Check if scikit-learn is installed..................[OK] (0.24.0)
Check if tensorflow (1.5.0) is installed............[OK] (1.5.0)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.55.0)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.26.2)
Check if pytest_console_scripts is installed........[OK]
Check if pytest-xdist is installed..................[OK] (2.2.0)
Check if tensorboard (1.14.0) is installed..........[WARNING] (2.4.0 != 1.14.0 mandated version))
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if DISPLAY variable is set....................[OK]
Check if figure can be opened with PyQt.............[OK]


Open a new Terminal window to load environment variables, or run:
source /home/kousu/.bashrc


Installation finished successfully!


real	4m41.226s
user	1m34.182s
sys	0m21.892s
```

</details>

<details><summary>without cleaning (-C): ~15s</summary>

``` 
$ time ./install_sct -Cy

 non-interactive mode



*******************************
* Welcome to SCT installation *
*******************************



Checking OS type and version...

Linux requiem 5.9.14-arch1-1 #1 SMP PREEMPT Sat, 12 Dec 2020 14:37:12 +0000 x86_64 GNU/Linux

Checking requirements...


OK!


SCT version ......... dev
Installation type ... in-place
Operating system .... linux (unknown)
Shell config ........ /home/kousu/.bashrc

--> Crash reports will not be sent.


SCT will be installed here: [/home/kousu/src/neuropoly/spinalcordtoolbox]


Skipping copy of source files (source and destination folders are the same)


Removing sct and isct softlink from /home/kousu/src/neuropoly/spinalcordtoolbox/bin


 conda will not be reinstalled


 venv_sct will not be recreated


Installing Python dependencies...


Using requirements.txt (git installation)

Looking in links: https://download.pytorch.org/whl/cpu/torch_stable.html
Ignoring torch: markers 'sys_platform == "darwin"' don't match your environment
Ignoring torchvision: markers 'sys_platform == "darwin"' don't match your environment
Requirement already satisfied: colored in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 1)) (1.4.2)
Requirement already satisfied: dipy in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 2)) (1.3.0)
Requirement already satisfied: futures in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 3)) (3.1.1)
Requirement already satisfied: h5py~=2.10.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 6)) (2.10.0)
Requirement already satisfied: Keras==2.1.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 7)) (2.1.5)
Requirement already satisfied: ivadomed==2.6.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 8)) (2.6.1)
Requirement already satisfied: matplotlib in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 9)) (3.3.3)
Requirement already satisfied: nibabel in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 10)) (3.2.1)
Requirement already satisfied: numpy in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 11)) (1.19.4)
Requirement already satisfied: onnxruntime==1.4.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 15)) (1.4.0)
Requirement already satisfied: pandas in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 16)) (1.1.5)
Requirement already satisfied: psutil in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 17)) (5.8.0)
Requirement already satisfied: pyqt5==5.11.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 18)) (5.11.3)
Requirement already satisfied: pytest in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 19)) (6.2.1)
Requirement already satisfied: pytest-cov in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 20)) (2.10.1)
Requirement already satisfied: raven in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 21)) (6.10.0)
Requirement already satisfied: requests in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 22)) (2.25.1)
Requirement already satisfied: requirements-parser in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 23)) (0.2.0)
Requirement already satisfied: scipy in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 24)) (1.5.4)
Requirement already satisfied: scikit-image in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 25)) (0.17.2)
Requirement already satisfied: scikit-learn in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 26)) (0.24.0)
Requirement already satisfied: tensorflow==1.5.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 27)) (1.5.0)
Requirement already satisfied: torch==1.5.0+cpu in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 33)) (1.5.0+cpu)
Requirement already satisfied: torchvision==0.6.0+cpu in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 35)) (0.6.0+cpu)
Requirement already satisfied: xlwt in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 37)) (1.3.0)
Requirement already satisfied: tqdm in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 38)) (4.55.0)
Requirement already satisfied: transforms3d in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 39)) (0.3.1)
Requirement already satisfied: urllib3[secure] in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 40)) (1.26.2)
Requirement already satisfied: pytest_console_scripts in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 41)) (1.1.0)
Requirement already satisfied: pytest-xdist in ./python/envs/venv_sct/lib/python3.6/site-packages (from -r requirements.txt (line 42)) (2.2.0)
Requirement already satisfied: seaborn in ./python/envs/venv_sct/lib/python3.6/site-packages (from ivadomed==2.6.1->-r requirements.txt (line 8)) (0.11.1)
Requirement already satisfied: bids-neuropoly>=0.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from ivadomed==2.6.1->-r requirements.txt (line 8)) (0.2)
Requirement already satisfied: pybids>=0.12.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from ivadomed==2.6.1->-r requirements.txt (line 8)) (0.12.4)
Requirement already satisfied: pillow>=7.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from ivadomed==2.6.1->-r requirements.txt (line 8)) (8.0.1)
Requirement already satisfied: joblib in ./python/envs/venv_sct/lib/python3.6/site-packages (from ivadomed==2.6.1->-r requirements.txt (line 8)) (1.0.0)
Requirement already satisfied: six in ./python/envs/venv_sct/lib/python3.6/site-packages (from h5py~=2.10.0->-r requirements.txt (line 6)) (1.15.0)
Requirement already satisfied: pyyaml in ./python/envs/venv_sct/lib/python3.6/site-packages (from Keras==2.1.5->-r requirements.txt (line 7)) (5.3.1)
Requirement already satisfied: protobuf in ./python/envs/venv_sct/lib/python3.6/site-packages (from onnxruntime==1.4.0->-r requirements.txt (line 15)) (3.14.0)
Requirement already satisfied: PyQt5_sip<4.20,>=4.19.11 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pyqt5==5.11.3->-r requirements.txt (line 18)) (4.19.19)
Collecting tensorboard==1.14.0
  Using cached tensorboard-1.14.0-py3-none-any.whl (3.1 MB)
Requirement already satisfied: absl-py>=0.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (0.11.0)
Requirement already satisfied: grpcio>=1.6.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (1.34.0)
Requirement already satisfied: wheel>=0.26 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (0.36.2)
Requirement already satisfied: setuptools>=41.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (51.0.0.post20201207)
Requirement already satisfied: werkzeug>=0.11.15 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (1.0.1)
Requirement already satisfied: markdown>=2.6.8 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard==1.14.0->-r requirements.txt (line 43)) (3.3.3)
Requirement already satisfied: future in ./python/envs/venv_sct/lib/python3.6/site-packages (from torch==1.5.0+cpu->-r requirements.txt (line 33)) (0.18.2)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from matplotlib->-r requirements.txt (line 9)) (2.4.7)
Requirement already satisfied: cycler>=0.10 in ./python/envs/venv_sct/lib/python3.6/site-packages (from matplotlib->-r requirements.txt (line 9)) (0.10.0)
Requirement already satisfied: python-dateutil>=2.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from matplotlib->-r requirements.txt (line 9)) (2.8.1)
Requirement already satisfied: kiwisolver>=1.0.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from matplotlib->-r requirements.txt (line 9)) (1.3.1)
Requirement already satisfied: threadpoolctl>=2.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from scikit-learn->-r requirements.txt (line 26)) (2.1.0)
Requirement already satisfied: packaging>=14.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from nibabel->-r requirements.txt (line 10)) (20.8)
Requirement already satisfied: pytz>=2017.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pandas->-r requirements.txt (line 16)) (2020.5)
Requirement already satisfied: importlib-metadata in ./python/envs/venv_sct/lib/python3.6/site-packages (from markdown>=2.6.8->tensorboard==1.14.0->-r requirements.txt (line 43)) (3.3.0)
Requirement already satisfied: patsy in ./python/envs/venv_sct/lib/python3.6/site-packages (from pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (0.5.1)
Requirement already satisfied: click in ./python/envs/venv_sct/lib/python3.6/site-packages (from pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (7.1.2)
Requirement already satisfied: bids-validator in ./python/envs/venv_sct/lib/python3.6/site-packages (from pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (1.5.8)
Requirement already satisfied: sqlalchemy<1.4.0.dev0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (1.3.22)
Requirement already satisfied: num2words in ./python/envs/venv_sct/lib/python3.6/site-packages (from pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (0.5.10)
Collecting tensorflow-tensorboard<1.6.0,>=1.5.0
  Using cached tensorflow_tensorboard-1.5.1-py3-none-any.whl (3.0 MB)
Requirement already satisfied: html5lib==0.9999999 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow==1.5.0->-r requirements.txt (line 27)) (0.9999999)
Requirement already satisfied: bleach==1.5.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow==1.5.0->-r requirements.txt (line 27)) (1.5.0)
Requirement already satisfied: pluggy<1.0.0a1,>=0.12 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 19)) (0.13.1)
Requirement already satisfied: iniconfig in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 19)) (1.1.1)
Requirement already satisfied: attrs>=19.2.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 19)) (20.3.0)
Requirement already satisfied: toml in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 19)) (0.10.2)
Requirement already satisfied: py>=1.8.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest->-r requirements.txt (line 19)) (1.10.0)
Requirement already satisfied: typing-extensions>=3.6.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard==1.14.0->-r requirements.txt (line 43)) (3.7.4.3)
Requirement already satisfied: zipp>=0.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard==1.14.0->-r requirements.txt (line 43)) (3.4.0)
Requirement already satisfied: mock>=2.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest_console_scripts->-r requirements.txt (line 41)) (4.0.3)
Requirement already satisfied: coverage>=4.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest-cov->-r requirements.txt (line 20)) (5.3.1)
Requirement already satisfied: execnet>=1.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest-xdist->-r requirements.txt (line 42)) (1.7.1)
Requirement already satisfied: pytest-forked in ./python/envs/venv_sct/lib/python3.6/site-packages (from pytest-xdist->-r requirements.txt (line 42)) (1.3.0)
Requirement already satisfied: apipkg>=1.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from execnet>=1.1->pytest-xdist->-r requirements.txt (line 42)) (1.5)
Requirement already satisfied: chardet<5,>=3.0.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests->-r requirements.txt (line 22)) (4.0.0)
Requirement already satisfied: idna<3,>=2.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests->-r requirements.txt (line 22)) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests->-r requirements.txt (line 22)) (2020.12.5)
Requirement already satisfied: tifffile>=2019.7.26 in ./python/envs/venv_sct/lib/python3.6/site-packages (from scikit-image->-r requirements.txt (line 25)) (2020.9.3)
Requirement already satisfied: imageio>=2.3.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from scikit-image->-r requirements.txt (line 25)) (2.9.0)
Requirement already satisfied: networkx>=2.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from scikit-image->-r requirements.txt (line 25)) (2.5)
Requirement already satisfied: PyWavelets>=1.1.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from scikit-image->-r requirements.txt (line 25)) (1.1.1)
Requirement already satisfied: decorator>=4.3.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from networkx>=2.0->scikit-image->-r requirements.txt (line 25)) (4.4.2)
Requirement already satisfied: pyOpenSSL>=0.14 in ./python/envs/venv_sct/lib/python3.6/site-packages (from urllib3[secure]->-r requirements.txt (line 40)) (20.0.1)
Requirement already satisfied: cryptography>=1.3.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from urllib3[secure]->-r requirements.txt (line 40)) (3.3.1)
Requirement already satisfied: cffi>=1.12 in ./python/envs/venv_sct/lib/python3.6/site-packages (from cryptography>=1.3.4->urllib3[secure]->-r requirements.txt (line 40)) (1.14.4)
Requirement already satisfied: pycparser in ./python/envs/venv_sct/lib/python3.6/site-packages (from cffi>=1.12->cryptography>=1.3.4->urllib3[secure]->-r requirements.txt (line 40)) (2.20)
Requirement already satisfied: docopt>=0.6.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from num2words->pybids>=0.12.4->ivadomed==2.6.1->-r requirements.txt (line 8)) (0.6.2)
Installing collected packages: tensorflow-tensorboard, tensorboard
  Attempting uninstall: tensorboard
    Found existing installation: tensorboard 2.4.0
    Uninstalling tensorboard-2.4.0:
      Successfully uninstalled tensorboard-2.4.0
Successfully installed tensorboard-1.14.0 tensorflow-tensorboard-1.5.1
Found existing installation: tensorflow-tensorboard 1.5.1
Uninstalling tensorflow-tensorboard-1.5.1:
  Successfully uninstalled tensorflow-tensorboard-1.5.1
Found existing installation: tensorboard 1.14.0
Uninstalling tensorboard-1.14.0:
  Successfully uninstalled tensorboard-1.14.0
Collecting tensorboard
  Using cached tensorboard-2.4.0-py3-none-any.whl (10.6 MB)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.7.0)
Requirement already satisfied: grpcio>=1.24.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.34.0)
Requirement already satisfied: markdown>=2.6.8 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (3.3.3)
Requirement already satisfied: protobuf>=3.6.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (3.14.0)
Requirement already satisfied: requests<3,>=2.21.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (2.25.1)
Requirement already satisfied: werkzeug>=0.11.15 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.0.1)
Requirement already satisfied: wheel>=0.26 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (0.36.2)
Requirement already satisfied: absl-py>=0.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (0.11.0)
Requirement already satisfied: google-auth<2,>=1.6.3 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.24.0)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (0.4.2)
Requirement already satisfied: six>=1.10.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.15.0)
Requirement already satisfied: setuptools>=41.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (51.0.0.post20201207)
Requirement already satisfied: numpy>=1.12.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from tensorboard) (1.19.4)
Requirement already satisfied: rsa<5,>=3.1.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard) (4.6)
Requirement already satisfied: pyasn1-modules>=0.2.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard) (0.2.8)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from google-auth<2,>=1.6.3->tensorboard) (4.2.0)
Requirement already satisfied: requests-oauthlib>=0.7.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard) (1.3.0)
Requirement already satisfied: importlib-metadata in ./python/envs/venv_sct/lib/python3.6/site-packages (from markdown>=2.6.8->tensorboard) (3.3.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in ./python/envs/venv_sct/lib/python3.6/site-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard) (0.4.8)
Requirement already satisfied: idna<3,>=2.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (2020.12.5)
Requirement already satisfied: chardet<5,>=3.0.2 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (4.0.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests<3,>=2.21.0->tensorboard) (1.26.2)
Requirement already satisfied: oauthlib>=3.0.0 in ./python/envs/venv_sct/lib/python3.6/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard) (3.1.0)
Requirement already satisfied: zipp>=0.5 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard) (3.4.0)
Requirement already satisfied: typing-extensions>=3.6.4 in ./python/envs/venv_sct/lib/python3.6/site-packages (from importlib-metadata->markdown>=2.6.8->tensorboard) (3.7.4.3)
Installing collected packages: tensorboard
Successfully installed tensorboard-2.4.0

Installing spinalcordtoolbox...

Obtaining file:///home/kousu/src/neuropoly/spinalcordtoolbox
Installing collected packages: spinalcordtoolbox
  Attempting uninstall: spinalcordtoolbox
    Found existing installation: spinalcordtoolbox dev
    Uninstalling spinalcordtoolbox-dev:
      Successfully uninstalled spinalcordtoolbox-dev
  Running setup.py develop for spinalcordtoolbox
Successfully installed spinalcordtoolbox

Creating launchers for Python scripts...


 SCT binaries will not be reinstalled


All requirements installed!


Installing data...


mkdir -p /home/kousu/src/neuropoly/spinalcordtoolbox/data


 data/PAM50 will not be reinstalled


 data/gm_model will not be reinstalled


 data/optic_models will not be reinstalled


 data/pmj_models will not be reinstalled


 data/deepseg_sc_models will not be reinstalled


 data/deepseg_gm_models will not be reinstalled


 data/deepseg_lesion_models will not be reinstalled


 data/c2c3_disc_models will not be reinstalled


 deepseg models will not be reinstalled


Validate installation...


--
Spinal Cord Toolbox (git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*)

sct_check_dependencies 
--

SCT info:
- version: git-ng/install-clean-optional-4fb20dd4e73ebf0a4d16dbd394907522be76b876*
- path: /home/kousu/src/neuropoly/spinalcordtoolbox
OS: linux (Linux-5.9.14-arch1-1-x86_64-with-arch)
CPU cores: Available: 4, Used by ITK functions: 4
RAM: Total: 15710MB, Used: 1884MB, Available: 12014MB
Check Python executable.............................[OK]
  Using bundled python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56) 
[GCC 7.3.0] at /home/kousu/src/neuropoly/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.2)
Check if dipy is installed..........................[OK] (1.3.0)
Check if futures is installed.......................[OK]
Check if h5py is installed..........................[OK] (2.10.0)
Check if Keras (2.1.5) is installed.................[OK] (2.1.5)
Check if ivadomed (2.6.1) is installed..............[OK] (2.6.1)
Check if matplotlib is installed....................[OK] (3.3.3)
Check if nibabel is installed.......................[OK] (3.2.1)
Check if numpy is installed.........................[OK] (1.19.4)
Check if onnxruntime (1.4.0) is installed...........[OK] (1.4.0)
Check if pandas is installed........................[OK] (1.1.5)
Check if psutil is installed........................[OK] (5.8.0)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.2.1)
Check if pytest-cov is installed....................[OK] (__version__ = '2.10.1')
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.25.1)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.5.4)
Check if scikit-image is installed..................[OK] (0.17.2)
Check if scikit-learn is installed..................[OK] (0.24.0)
Check if tensorflow (1.5.0) is installed............[OK] (1.5.0)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.55.0)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.26.2)
Check if pytest_console_scripts is installed........[OK]
Check if pytest-xdist is installed..................[OK] (2.2.0)
Check if tensorboard (1.14.0) is installed..........[WARNING] (2.4.0 != 1.14.0 mandated version))
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if DISPLAY variable is set....................[OK]
Check if figure can be opened with PyQt.............[OK]


Open a new Terminal window to load environment variables, or run:
source /home/kousu/.bashrc


Installation finished successfully!


real	0m16.180s
user	0m14.878s
sys	0m3.010s

```

</details>

Reply:
(of course if you actually had changed some python deps it would be a bit longer; but not 4 minutes longer)

Reply:
> > Now, we might have some trouble with this at first because we don't pin our versions
> 
> Could you clarify what trouble you think we might run into? I might be off-base with being against version-pinning.

`pip install whatever` isn't that reliable about picking what should be installed. If `whatever` is already installed, regardless of version, pip keeps it, but if not, it gets the newest. So reviving an old venv that hasn't been touched in a while will probably break, which is why just wiping it and starting fresh is safer.

> > I could go through an add ~= to the current requirements.txt based on what versions we currently use as part of this PR. That would be a step towards making #1526 work.
> 
> I shared some related thoughts about using `~=` here: [neuropoly/axondeepseg#411 (comment)](https://github.com/neuropoly/axondeepseg/issues/411#issuecomment-740145463). At the time, I was actually against using `~=` for everything, in favor of keeping our development requirements more open.

I think `~=` keeps the requirements open to upgrades, just not major version upgrades, because those are _supposed_ to mean incompatibilities you need to manually check for.

The rules are..kind of involved though: https://www.python.org/dev/peps/pep-0440/#examples

>     ~=3.1: version 3.1 or later, but not version 4.0 or later.
>     ~=3.1.2: version 3.1.2 or later, but not version 3.2.0 or later.

The first example makes sense to me, but not the second. I would have thought ~=3.1.2 meant >=3.1.2, ==3.*, i.e. >=3.1.2, <4.0.0, but it's not, it's basically >=3.1.2,<3.3.0. I think I'm confused because pip **doesn't** use semver: https://www.python.org/dev/peps/pep-0440/#semantic-versioning, though it claims to be compatible with it.

What if we started out just match the major version, then:

```
torch~=1.5
```

if there's a fix we absolutely need in 1.5.6:

```
torch~=1.5,>=1.5.6
```

or maybe we should just stick to

```
torch>=1.5,<2
```

since that's clearer; if we need a patch there then just change it to:


```
torch>=1.5.6,<2
```